geoportailv3/static/js/layerinfo/layerinfoservice.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * @fileoverview Provides a layer info service. That service is a function used
+ * to retrieve the info (metadata) for a layer.
+ */
+
+goog.provide('app.ShowLayerinfo');
+
+goog.require('ngeo.CreatePopup');
+
+/**
+ * @typedef {function(ol.layer.Layer)}
+ */
+app.ShowLayerinfo;
+
+
+/**
+ * @param {angular.$http} $http Angular $http service
+ * @param {angular.$sce} $sce Angular $sce service
+ * @param {ngeo.CreatePopup} ngeoCreatePopup Ngeo popup factory service
+ * @ngInject
+ */
+app.showLayerinfoFactory = function($http, $sce, ngeoCreatePopup) {
+
+  /**
+   * @type {ngeo.Popup}
+   */
+  var popup = ngeoCreatePopup();
+
+  return (
+
+      /**
+       * @param {ol.layer.Layer} layer The layer
+       */
+      function(layer) {
+        popup.show();
+        var title = /** @type {string} */ (layer.get('label'));
+        popup.setTitle(title);
+        var content = /** @type {string} */
+            ($sce.trustAsHtml("<b>some</b> content"));
```

Comment:
On Mon, Feb 2, 2015 at 5:43 PM, Pierre GIRAUD <notifications@github.com> wrote:
>
> In geoportailv3/static/js/layerinfo/layerinfoservice.js:
>
> > +  /**
> > +   * @type {ngeo.Popup}
> > +   */
> > +  var popup = ngeoCreatePopup();
> > +
> > +  return (
> > +
> > +      /**
> > +       * @param {ol.layer.Layer} layer The layer
> > +       */
> > +      function(layer) {
> > +        popup.show();
> > +        var title = /** @type {string} */ (layer.get('label'));
> > +        popup.setTitle(title);
> > +        var content = /** @type {string} */
> > +            ($sce.trustAsHtml("<b>some</b> content"));
>
> No. It's a bit oscure to me.


Actually, $sce.trustAsHtml does not return a string. It returns an
object that can be used with ng-bind-html. So casting to a string does
not really make sense. The problem is in ngeo, in the signature of the
setContent function.



-- 
Eric Lemoine

Camptocamp France SAS
Savoie Technolac, BP 352
73377 Le Bourget du Lac, Cedex

Tel : 00 33 4 79 44 44 94
Mail : eric.lemoine@camptocamp.com
http://www.camptocamp.com